From f190b6f8506745170cbb3754f21ab36267847483 Mon Sep 17 00:00:00 2001
From: Thomas Gleixner <tglx@linutronix.de>
Date: Tue, 21 Jul 2009 16:07:37 +0200
Subject: [PATCH 096/267] genirq: Disable random call on preempt-rt

The random call introduces high latencies and is almost
unused. Disable it for -rt.

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 kernel/irq/handle.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/kernel/irq/handle.c b/kernel/irq/handle.c
index 470d08c..634620c 100644
--- a/kernel/irq/handle.c
+++ b/kernel/irq/handle.c
@@ -156,8 +156,11 @@ handle_irq_event_percpu(struct irq_desc *desc, struct irqaction *action)
 		action = action->next;
 	} while (action);
 
+#ifndef CONFIG_PREEMPT_RT_FULL
+	/* FIXME: Can we unbreak that ? */
 	if (random & IRQF_SAMPLE_RANDOM)
 		add_interrupt_randomness(irq);
+#endif
 
 	if (!noirqdebug)
 		note_interrupt(irq, desc, retval);
-- 
1.7.10

